File: html/redcap_v15.7.6/API/examples/ruby/rename_record.rb
#!/usr/bin/env ruby
require 'json'
require 'curl'
require './settings.rb'
include Settings

fields = {
    :token => Settings::API_TOKEN,
    :action => 'rename',
    :content => 'record',
    :record => '1',
    :new_record_name => 'record_1',
    :arm => '1',
    :returnFormat => 'json'
}
ch = Curl::Easy.http_post(
   Settings::API_URL,
  fields.collect{|k, v| Curl::PostField.content(k.to_s, v)}
)
puts ch.body_str

File: html/redcap_v15.7.6/API/examples/perl/rename_record.pl
#!/usr/bin/env perl
use strict;
use warnings;
use LWP::Curl;

my %config = do 'config.pl';
my $data = {
    token => $config{api_token},
    action => 'rename',
    content => 'record',
    record => '1',
    new_record_name => 'record_1',
    arm => '1',
    returnFormat => 'json'
};
my $ch = LWP::Curl->new(auto_encode => 0);
my $content = $ch->post(
    $config{api_url},
    $data,
    $config{referer}
);
print $content;